### PR TITLE
Add repeatable deployment for service fabric

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

File: src/Microsoft.DncEng.DeployServiceFabricCluster/Program.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Mono.Options;
+
+namespace Microsoft.DncEng.DeployServiceFabricCluster
+{
+    class HostEnvironment : IHostEnvironment
```
Comment (ChadNedzlek) [inline src/Microsoft.DncEng.DeployServiceFabricCluster/Program.cs]: public?

File: src/Microsoft.DncEng.DeployServiceFabricCluster/Program.cs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            if (string.IsNullOrEmpty(parameter))
+            {
+                Fatal(2, $"the {name} parameter is required.");
+            }
+        }
+
+        [DoesNotReturn]
+        private static void Fatal(int code, string errorMessage)
+        {
+            Console.Error.WriteLine("fatal: " + errorMessage);
```
Comment (ChadNedzlek) [inline src/Microsoft.DncEng.DeployServiceFabricCluster/Program.cs]: Would it be helpful to VSTS error this up?

File: src/Microsoft.DncEng.DeployServiceFabricCluster/Program.cs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+            RequireParameter(configFile, "--config");
+            RequireParameter(environment, "--environment");
+
+            if (!File.Exists(configFile))
+            {
+                Fatal(3, $"config file '{configFile}' does not exist.");
+            }
+
+            var configFileName = Path.GetFileNameWithoutExtension(configFile);
```
Comment (ChadNedzlek) [inline src/Microsoft.DncEng.DeployServiceFabricCluster/Program.cs]: Lot of var, my good sir.

File: src/Microsoft.DncEng.DeployServiceFabricCluster/Program.cs (hunk 2 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.DeployServiceFabricCluster/Program.cs]: yep, ill add that

File: src/Microsoft.DncEng.DeployServiceFabricCluster/Program.cs (hunk 3 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.DeployServiceFabricCluster/Program.cs]: will fix.

**Pull request merged**